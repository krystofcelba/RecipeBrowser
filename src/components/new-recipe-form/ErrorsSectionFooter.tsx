import React from 'react';

import { View, LeftPaddedErrorText, LeftPaddedBoldErrorText } from 'src/components/common';
import AppTheme from 'src/assets/appTheme';
import { isNumber } from 'src/utils';

interface Props {
  errors: {};
  testIDcontext: string;
}

const ErrorsSectionFooter = ({ errors, testIDcontext }: Props) => (
  <View paddingLeft={AppTheme.defaultPadding / 2} paddingTop={AppTheme.defaultPadding / 2}>
    {Object.keys(errors).map(key => {
      const error = errors[key];
      return error ? (
        <View testID={`${testIDcontext}:error:${key}`} flexDirection="row" key={key}>
          <LeftPaddedBoldErrorText>{isNumber(key) ? `#${Number(key) + 1}` : '\u2022'}</LeftPaddedBoldErrorText>
          <LeftPaddedErrorText>{error}</LeftPaddedErrorText>
        </View>
      ) : null;
    })}
  </View>
);

export default ErrorsSectionFooter;
