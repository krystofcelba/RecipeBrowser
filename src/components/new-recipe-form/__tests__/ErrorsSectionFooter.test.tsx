import React from 'react';
import { shallow } from 'enzyme';

import ErrorsSectionFooter from '../ErrorsSectionFooter';

describe('ValidationErrors component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<ErrorsSectionFooter errors={{}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected with defined errors', () => {
    const wrapper = shallow(<ErrorsSectionFooter errors={{ name: 'error', description: 'error1', test: undefined }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected with defined errors for arrays', () => {
    const wrapper = shallow(<ErrorsSectionFooter errors={{ '0': 'array error 0', '1': 'array error 1' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
