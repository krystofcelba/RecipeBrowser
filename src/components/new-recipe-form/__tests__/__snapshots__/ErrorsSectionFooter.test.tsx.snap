// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationErrors component renders as expected 1`] = `
<glamorous.View
  paddingLeft={5}
  paddingTop={5}
/>
`;

exports[`ValidationErrors component renders as expected with defined errors 1`] = `
<glamorous.View
  paddingLeft={5}
  paddingTop={5}
>
  <glamorous.View
    flexDirection="row"
    key="name"
  >
    <LeftPaddedBoldErrorText>
      •
    </LeftPaddedBoldErrorText>
    <ErrorText>
      error
    </ErrorText>
  </glamorous.View>
  <glamorous.View
    flexDirection="row"
    key="description"
  >
    <LeftPaddedBoldErrorText>
      •
    </LeftPaddedBoldErrorText>
    <ErrorText>
      error1
    </ErrorText>
  </glamorous.View>
</glamorous.View>
`;

exports[`ValidationErrors component renders as expected with defined errors for arrays 1`] = `
<glamorous.View
  paddingLeft={5}
  paddingTop={5}
>
  <glamorous.View
    flexDirection="row"
    key="0"
  >
    <LeftPaddedBoldErrorText>
      #1
    </LeftPaddedBoldErrorText>
    <ErrorText>
      array error 0
    </ErrorText>
  </glamorous.View>
  <glamorous.View
    flexDirection="row"
    key="1"
  >
    <LeftPaddedBoldErrorText>
      #2
    </LeftPaddedBoldErrorText>
    <ErrorText>
      array error 1
    </ErrorText>
  </glamorous.View>
</glamorous.View>
`;
